### ISSUE TITLE
truncate characters, not words?

### ISSUE THREAD

**Issue by tunnckoCore** [helpers/helper-related#9]
It would be better. That's ugly.

![screenshot from 2016-03-18 05-40-30](https://cloud.githubusercontent.com/assets/5038030/13867872/10e800ca-eccc-11e5-8439-bf0322c61980.png)

Almost every time I pass different number to options in `.verb.md`, because this going to new line is shit for me.

In this case exactly `get-fn-name` can have more 3 words to the end of line, so 14 - currently the limit is 11, my default is 13 and lib's limit is 15.

In short, it's ugly and cost time and nerves.

edit: And all is because this "homepage" thing, which can be just the name of the package and no more shits. But yea, I understand you and your ideas and why you do it, I also worked in SEO, but don't like it here.

**Comment by jonschlinkert** [comment]
> In short, it's ugly and cost time and nerves.

holy moses, take a deep breath lol. we'll get through this together! 

how about if we make the links into templates, so that helpers can be used on the urls? 

something like (pseudo-code, obviously you can create custom helpers to do whatever you want here):

``` html
<%= link(name) %>: <%= truncate(description, 13) %> etc etc
```

all we need to do is have the helper return an object instead of a string.

**Comment by tunnckoCore** [comment]
> holy moses, take a deep breath lol.

hahaha, yea, the situation isn't so bad, lol.

> you can create custom helpers to do whatever you want here

yes, but in anyway it will be in defaults, why to add more helpers? This also will require verbfile everywhere.

Good idea may be to have template option and also this helper to try to load default template otherwise from global config.

**Comment by jonschlinkert** [comment]
> This also will require verbfile everywhere.

no, you just define them in verb config in package.json :)

**Comment by tunnckoCore** [comment]
> you just define them in verb config

What? The helpers can be defined in verb config? If yes... huh, cool.

**Comment by jonschlinkert** [comment]
yeah, it should work. if not it would be a bug at this point

edit: in package.json, just do

``` json
{
  "verb": {
    "helpers": ["helper-foo"]
  }
}
```

You might also be able to define the helpers as objects, where the helper name is a key on the `helpers` object, so you can pass options etc. but I don't remember if I implemented that yet

**Comment by jonschlinkert** [comment]
also, make sure the helper is in `devDependencies`

**Comment by jonschlinkert** [comment]
here are the properties that have been implemented in `verb` config: https://github.com/node-base/base-runner/tree/master/lib/config

there is a schema that normalizes each value before being processed by `app.config.process()`, so each property has a corresponding `field` on the schema: https://github.com/node-base/base-runner/blob/master/lib/fields/helpers.js

**Comment by tunnckoCore** [comment]
Huh, okey :+1: Cool. I'll think what I can do for me. :)

**Comment by tunnckoCore** [comment]
Hm... I think I have an idea. I started new simple related helper and found great [ellipsize](https://github.com/mvhenten/ellipsize) package which counts characters and truncate to nearest space and started playing. And all looks pretty good without ugliness.

Github readme width allows between 90 and 110 characters at row, it depends on if it has `more`, `homepage` and the length of the name of the package.

And i've implemented this loop which is looping over getPkgs.

``` js
    while (i < len) {
        var pkg = pkgs[i++]
        var url = getUrl(pkg)
        var npmLink = 'https://www.npmjs.com/package/' + pkg.name

        res += '- ' + link(pkg.name, npmLink)
        if (pkg.description && pkg.description.length > options.truncate) {
          var truncate = pkg.name.length > 5 ? options.truncate - pkg.name.length : options.truncate
          res += ': ' + ellipsize(pkg.description || '', truncate)
          res += link('more', npmLink) + ' | ' + link('homepage', url)
        } else {
          res += pkg.description ? ': ' + pkg.description : ''
        }

        res += '\n'
      }
      cb(null, res)


function link(anchor, href) {
  return '[' + anchor + '](' + href + ')';
}

function getUrl (pkg) {
  if (pkg.homepage) return pkg.homepage.replace('#readme', '')
  if (typeof pkg.repository === 'string') return 'https://github.com/' + pkg.repository
  if (typeof pkg.repository === 'object' && pkg.repository.url) return pkg.repository.url
  return 'https://www.npmjs.com/package/' + pkg.name
}
```

and outputs everything as it is currently (package of the name and `more` are links to npm, only `homepage` is link to the actual homepage/github repo), but the little trick is that `if` check over truncate. It downgrades truncate option length with the name of the package. Default truncate is `90` characters (to can set `more` and `homepage` in same row).
If package name length is bigger than 5 chars it `options.truncate - pkg.name.length`.

So given few packages it outputs:

![screenshot from 2016-03-19 01-36-38](https://cloud.githubusercontent.com/assets/5038030/13894649/2c6601c4-ed73-11e5-8a0b-4dc89e24db21.png)

and it never won't go on next line, it's adjustable and dynamic, no matter how long is the name, how long is the description - it always will set everything on one line.

I'll try to do a PR if you want. And please go review the `ellipsize` in a minute, it looks as what exactly need to do the things correctly and beautiful. Maybe you or I can think to transform it to work this way for words, but I think on per word basis is harder. It's not harder, but it's already done.

edit: also it won't fail if `pkg.homepage` is not set. Which `helper-related` currently fails

```
/home/charlike/.nvm/versions/node/v5.8.0/lib/node_modules/verb-readme-generator/node_modules/helper-related/index.js:132
  res += link('homepage', config.homepage.replace(/#readme$/, ''));
                                         ^

TypeError: Cannot read property 'replace' of undefined
    at toLink (/home/charlike/.nvm/versions/node/v5.8.0/lib/node_modules/verb-readme-generator/node_modules/helper-related/index.js:132:42)

```

i think because `karma-server-side`

### helper-related result

![screenshot from 2016-03-19 01-41-35](https://cloud.githubusercontent.com/assets/5038030/13894717/e1285990-ed73-11e5-92b2-e5961759e22d.png)
